modules/payloads/adapters/cmd/windows/python.rb

```diff
@@ -0,0 +1,50 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+module MetasploitModule
+  include Msf::Payload::Adapter
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Python Exec',
+        'Description' => 'Execute a Python payload from a command',
+        'Author' => 'Spencer McIntyre',
+        'Platform' => 'win',
+        'Arch' => ARCH_CMD,
+        'License' => MSF_LICENSE,
+        'AdaptedArch' => ARCH_PYTHON,
+        'AdaptedPlatform' => 'python',
+        'RequiredCmd' => 'python'
+      )
+    )
+    register_advanced_options(
+      [
+        OptString.new('PythonPath', [true, 'The path to the Python executable', 'python'])
+      ]
+    )
+  end
+
+  def compatible?(mod)
+    # size is not unlimited due to the standard command length limit, the final size depends on the options that are
+    # configured but 3,000 is in a good range (can go up to 4,000 with default settings at this time)
+    if mod.type == Msf::MODULE_PAYLOAD && (mod.class.const_defined?(:CachedSize) && mod.class::CachedSize != :dynamic) && (mod.class::CachedSize >= 3_000)
+      return false
+    end
+
+    super
+  end
+
+  def generate
```

Comment:
Whilst we have the `generate` command here I'm also looking at `generate_stage` and `generate_payload_uuid` as well as `handle_connection` from the `adapters/cmd/windows/powershell.rb` file and I'm wondering if we should be adding those in here as well.

Reply:
That was added to fix the encrypted shells that needed the original architecture to be passed to the compiler. The Python payloads are simpler and don't reference the architecture. What you see here is more along the lines of https://github.com/rapid7/metasploit-framework/blob/master/modules/payloads/adapters/cmd/unix/python.rb.